src/interceptors/ClientRequest/utils/getIncomingMessageBody.ts

```diff
@@ -1,12 +1,20 @@
+import * as zlib from 'zlib'
 import { IncomingMessage } from 'http'
+import { Stream } from 'stream'
 
 export function getIncomingMessageBody(res: IncomingMessage): Promise<string> {
   let responseBody = ''
+  let stream: Stream = res
+
+  if (res.headers['content-encoding'] === 'gzip') {
+    const gzip = zlib.createGunzip()
```

Comment:
What implications do you see if we adopt your previous suggestion:

```js
let responseBody = Buffer.from('')
```

I'm hesitant to bring gzip handling to this method, it may be a stretch for its responsibilities.